layers/+chat/circe/packages.el

```diff
@@ -0,0 +1,151 @@
+  (use-package circe
+    :commands circe
+    :config
+
+    (set-face-background 'circe-prompt-face nil)
+    (setq circe-reduce-lurker-spam t)
+    (setq circe-use-cycle-completion t)
+    (setq circe-active-users-timeout (* 60 30)) ; 30 minutes
+    (setq circe-format-say "{nick}> {body}")
+    (setq circe-format-self-say (concat (propertize ">>>" 'face 'circe-self-say-face) " {body}"))
+    (setq circe-prompt-string (concat (propertize ">>>" 'face 'circe-prompt-face) " "))
+    (setq circe-highlight-nick-type 'all)
+
+    ;; Keybindings
+    (spacemacs/set-leader-keys "aic" 'circe)
+
+    (dolist (mode circe-modes)
+      (spacemacs/declare-prefix-for-mode mode "mc" "command")
+      (spacemacs/set-leader-keys-for-major-mode mode
+        "r" 'circe-reconnect
+        "R" 'circe-reconnect-all
+        "v" 'circe-version
+        "c n" 'circe-command-NAMES
+        "c q" 'circe-command-QUERY
+        "c h" 'circe-command-HELP
+        "c t" 'circe-command-CHTOPIC
+        "c s" 'circe-command-SAY
+        ))))
+
+(defun circe/init-circe-color-nicks ()
+    (use-package circe-color-nicks
+      :after circe
+      :commands enable-circe-color-nicks
+      :config
+      (setq circe-color-nicks-everywhere t)))
+
+(defun circe/init-lui ()
+  (spacemacs|use-package-add-hook circe
+    :post-config
+    (use-package lui
+      :commands lui-mode
+      :config
+      ;; Show channel name in prompt.
+      (add-hook 'circe-chat-mode-hook
+                '(let* ((chan
+                         ((s buffer-name)
+                          (_ gitter-fmt)
+                          (s-match (rx (+ nonl) "/" (group (+ nonl)) eos) s))
+                         ((_ standard-fmt)
+                          (s-match (rx "#" (group (+ nonl)) eos) s))
+                         (or gitter-fmt standard-fmt))
+                        (prompt (concat "#" (propertize chan 'face 'circe-prompt-face) " > ")))
+                   (lui-set-prompt prompt)))
+
+      ;; Timestamps in margins.
+      (setq lui-time-stamp-position 'right-margin)
+      (setq lui-time-stamp-format "%H:%M")
+      (setq lui-fill-type nil)
+      (add-hook 'lui-mode-hook (lambda ()
+                                 (setq fringes-outside-margins t)
+                                 (setq right-margin-width 5)
+                                 (setq word-wrap t)
+                                 (setq wrap-prefix "    "))))))
+
+(defun circe/init-lui-autopaste ()
+  (spacemacs|use-package-add-hook lui
+    :post-config
+    (use-package lui-autopaste
+      :commands enable-lui-autopaste
+      :init
+      (add-hook 'circe-channel-mode-hook #'enable-lui-autopaste))))
+
+(defun circe/post-init-helm ()
+   (with-eval-after-load 'helm
+     (if (not circe-helm-for-username-completion)
+         (dolist (mode circe-modes)
+           (add-to-list 'helm-mode-no-completion-in-region-in-modes mode)))))
+
+(defun circe/init-helm-circe ()
+  (use-package helm-circe
+    :defer t
```

Comment:
when you set `:defer t` it tells use-package that some other code is going to load the package. I don't think this is the case. Instead use the `:comands` keyword to tell use-package to load this package when certain commands are run, which will implicitly defer the package. So in this example replace `:defer t` with `:commands (helm-circe helm-circe-servers <etc>)`.